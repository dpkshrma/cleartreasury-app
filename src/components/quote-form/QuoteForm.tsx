import * as React from "react";
import Countdown from "../countdown/Countdown";
import { Button, MoneyInput } from "@clear-treasury/design-system";
import "flag-icon-css/css/flag-icon.css"; //This is imported in the select component in the design system - should be global?

export interface CurrencyOptionProps {
  label?: string;
  value?: string;
  icon?: React.ReactNode;
  isHighlighted?: boolean;
}

export interface QuoteFormProps {
  title: string;
  currencies: CurrencyOptionProps[];
}

type Ref = HTMLFormElement;

const QuoteForm = React.forwardRef<Ref, QuoteFormProps>((props) => {
  return (
    <div className="space-y-6 max-w-2xl px-4">
      {/* ^^ Max width and padding not 100% right. Might depend on parent container? */}
      <h1 className="block w-full text-theme-color-on-surface text-2xl">
        {props.title}
      </h1>
      <MoneyInput
        options={props.currencies}
        name="money_sent"
        label="You send"
      />
      <MoneyInput
        options={props.currencies}
        name="money_recieved"
        label="They recieve"
      />
      <div className="flex justify-between">
        <p className="text-lg text-theme-color-on-surface">Exchange rate</p>
        <div className="flex">
          {/* eslint-disable-next-line no-console */}
          <Countdown time={2} onComplete={() => console.log("Finished")} />
          <span className="ml-2 text-lg text-gray-400">1.001</span>
        </div>
      </div>
      <div className="flex justify-between space-x-4 border-t border-gray-200 py-6">
        <span className="text-sm text-gray-500">
          The rate quoted is a live rate, valid for 20 seconds. You will be
          asked to accept a confirmed rate as part of the transfer process.
        </span>
        <Button size={Button.Size.LARGE}>Continue</Button>
      </div>
    </div>
  );
});

export default QuoteForm;
