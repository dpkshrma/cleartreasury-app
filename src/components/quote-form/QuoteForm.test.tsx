import * as React from "react";
import { render } from "../../../test/testUtils";
import * as Stories from "./QuoteForm.stories";

describe("Quote Form", () => {
  it("matches snapshot", () => {
    const { asFragment } = render(
      <Stories.Default {...Stories.Default.args} />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
