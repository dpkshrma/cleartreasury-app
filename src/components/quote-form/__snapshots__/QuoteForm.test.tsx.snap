// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Quote Form matches snapshot 1`] = `
<DocumentFragment>
  <form
    class="space-y-6"
  >
    <h1
      class="block w-full text-theme-color-on-surface text-2xl"
    >
      How much would you like to transfer?
    </h1>
    <div
      class="w-full relative "
      role="group"
    >
      <span
        class="text-base text-gray-700 inline-block mb-2"
        id="money-input-label"
      >
        You send
      </span>
      <div
        class="flex items-stretch"
      >
        <div
          class="w-full relative "
        >
          <input
            aria-label="sell amount"
            class="form-input border border-gray-300 bg-white mb-2 p-3 w-full focus:ring-teal-100 focus:border-teal-300 disabled:cursor-not-allowed placeholder-gray-400 rounded-l"
            id="sell_amount"
            name="sell_amount"
            type="text"
            value="1000.00"
          />
        </div>
        <div
          class="w-40"
        >
          <label
            class="sr-only"
            id="sell_currency-label"
          >
            sell currency
          </label>
          <div
            class="w-full relative "
          >
            <button
              aria-expanded="false"
              aria-haspopup="listbox"
              aria-labelledby="sell_currency-label sell_currency-toggle-button"
              class="form-select border border-gray-300 focus:ring-teal-100 focus:border-teal-300 disabled:cursor-not-allowed w-full mb-2 p-3 text-left whitespace-nowrap text-theme-color-on-primary bg-theme-color-primary rounded-r"
              data-testid="sell_currency"
              id="sell_currency-toggle-button"
              name="sell_currency"
              type="button"
              value="GBP"
            >
              <div
                class="flex space-x-2"
              >
                <span
                  class="flag-icon-gb pl-6 bg-left bg-contain bg-no-repeat inline-block w-6 h-6"
                />
                <span>
                  GBP
                </span>
              </div>
            </button>
            <ul
              aria-labelledby="sell_currency-label"
              class="absolute max-h-56 mt-1 overflow-auto z-10 w-full"
              id="sell_currency-menu"
              role="listbox"
              tabindex="-1"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      class="w-full relative "
      role="group"
    >
      <span
        class="text-base text-gray-700 inline-block mb-2"
        id="money-input-label"
      >
        They recieve
      </span>
      <div
        class="flex items-stretch"
      >
        <div
          class="w-full relative "
        >
          <input
            aria-label="buy amount"
            class="form-input border border-gray-300 bg-white mb-2 p-3 w-full focus:ring-teal-100 focus:border-teal-300 disabled:cursor-not-allowed placeholder-gray-400 rounded-l"
            id="buy_amount"
            name="buy_amount"
            type="text"
            value=""
          />
        </div>
        <div
          class="w-40"
        >
          <label
            class="sr-only"
            id="buy_currency-label"
          >
            buy currency
          </label>
          <div
            class="w-full relative "
          >
            <button
              aria-expanded="false"
              aria-haspopup="listbox"
              aria-labelledby="buy_currency-label buy_currency-toggle-button"
              class="form-select border border-gray-300 focus:ring-teal-100 focus:border-teal-300 disabled:cursor-not-allowed w-full mb-2 p-3 text-left whitespace-nowrap text-theme-color-on-primary bg-theme-color-primary rounded-r"
              data-testid="buy_currency"
              id="buy_currency-toggle-button"
              name="buy_currency"
              type="button"
              value="EUR"
            >
              <div
                class="flex space-x-2"
              >
                <span
                  class="flag-icon-eu pl-6 bg-left bg-contain bg-no-repeat inline-block w-6 h-6"
                />
                <span>
                  EUR
                </span>
              </div>
            </button>
            <ul
              aria-labelledby="buy_currency-label"
              class="absolute max-h-56 mt-1 overflow-auto z-10 w-full"
              id="buy_currency-menu"
              role="listbox"
              tabindex="-1"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      class="flex justify-between"
    >
      <p
        class="text-lg text-theme-color-on-surface"
      >
        Exchange rate
      </p>
      <div
        class="flex space-x-4"
      >
        <span />
      </div>
    </div>
    <div
      class="flex justify-between space-x-4 border-t border-gray-200 py-6"
    >
      <span
        class="text-sm text-gray-500"
      >
        The rate quoted is a live rate, valid for 20 seconds. You will be asked to accept a confirmed rate as part of the transfer process.
      </span>
      <button
        class="inline-flex items-center justify-center gap-2
    border border-transparent
    ring-theme-button-primary
    text-center leading-snug
    rounded-theme-radius
    focus:outline-none focus:ring-1 focus:ring-offset-1 focus:ring-offset-transparent
    transition-all duration-300
    hover:shadow-md hover:cursor-pointer bg-theme-button-primary text-theme-color-on-primary hover:bg-theme-button-primary-hover py-3 px-8"
      >
        Continue
      </button>
    </div>
  </form>
</DocumentFragment>
`;
